### PR TITLE
Add Client Version Configuration

### PR DESCRIPTION
**Pull request by MChorfa**
Add an example of defining a client version for  a mixin if required.

### CODE REVIEW

File: pkg/skeletor/build.go [hunk 1]
```diff
@@ -15,6 +44,59 @@ package skeletor
...
+		err := yaml.Unmarshal(contents, &input)
+		return &input, err
+	})
+	if err != nil {
+		return err
+	}
+
+	suppliedClientVersion := input.Config.ClientVersion
+
+	if suppliedClientVersion != "" {
```
Comment (vdice) [inline pkg/skeletor/build.go:66]: I might suggest we remove the semver constraint.  I'm thinking for the most cases of mixins, semver restrictions won't be needed and may be prove too restrictive. WDYT?

Comment (MChorfa) [reply pkg/skeletor/build.go:66]: Yes, you are right! semver restrictions are too much and you are right also about the comment/uncomment code, looks pretty noisy. There is another way to generate the desired code, but we have to use `https://github.com/cookiecutter/cookiecutter`  and this mean this project  would be converted into `cookiecutter` style, and then we could much more flexibility of generating an initial mixin code

Comment (vdice) [reply pkg/skeletor/build.go:51]: The cookiecutter project looks interesting and perhaps worth revisiting in the future.

I think what we have now with your updates is great: support/example for processing mixin config, with clientVersion as a great standard entry.  I think if we add additional context/comments around the `//fmt.Fprintf(m.Out, dockerfileLines)` line, maybe showing an example that uses `m.ClientVersion`, we should be good to go.

File: pkg/skeletor/mixin.go [hunk 2]
```diff
@@ -5,15 +5,20 @@ import (
...
 	//add whatever other context/state is needed here
 }
 
 // New azure mixin client, initialized with useful defaults.
 func New() (*Mixin, error) {
 	return &Mixin{
-		Context: context.New(),
+		Context:       context.New(),
+		ClientVersion: defaultClientVersion,
+		//add whatever is needed here
```
Comment (vdice) [inline pkg/skeletor/mixin.go:21]: nit: I wonder if the similar note in the `Mixin` struct declaration is enough?  Perhaps we don't need the same note here as well?

**Comment by MChorfa** [comment]
Yes absolutely :) WDYT about this?:
You will find also `client version` basic logic ready to go, where you can support the `client version` definition for your mixing if required.

**Comment by vdice** [comment]
@MChorfa That sounds great!

**Comment by carolynvs** [comment]
WOOOOOOOOO! 🍪

**Pull request merged**